src/routers/mint/controller.ts

```diff
@@ -1,80 +1,122 @@
 import controller from "../controller";
-import { Request, Response } from "express";
-import {  checkUserThenCreateNftWF, getUserNftAfterCheck, getMagicLinkFromAirtableWF, getUserMagicLinkFromAirtable, getUserScore } from "../../workflows/mint/workflows";
+import { NextFunction, Request, Response } from "express";
+import {
+  checkUserThenCreateNftWF,
+  getUserNftAfterCheck,
+  getMagicLinkFromAirtableWF,
+  getUserMagicLinkFromAirtable,
+  getUserScore,
+} from "../../workflows/mint/workflows";
 import { plainToClass } from "class-transformer";
 import { MintDTO } from "../../models/mintDto";
-import { getEmailFromIdToken, getWalletPublicKeyFromIdToken } from "../../services/solana";
+import {
+  getEmailFromIdToken,
+  getWalletPublicKeyFromIdToken,
+} from "../../services/solana";
 import { createTemporalClient } from "../../services/temporal";
 import { UserDTO } from "../../models/userDto";
-
+import {
+  WorkflowExecutionAlreadyStartedError,
+  WorkflowIdReusePolicy,
+} from "@temporalio/client";
+import { nextTick } from "process";
 
 class MintController extends controller {
-  
-  checkUserThenCreateNft = async (req: Request, res: Response) => {
-    const idToken = req.headers["id-token"]!;
-    // const userEmail = await getEmailFromIdToken(idToken.toString());
-    const userDTO = new UserDTO
-    userDTO.publicKey = await getWalletPublicKeyFromIdToken(idToken.toString());
-    userDTO.email = await getEmailFromIdToken(idToken.toString());
-    const client = await createTemporalClient();
-    const workFlowId = "user-" + req.body.wfId;
-    userDTO.wfId = workFlowId;
-    const handle = await client.workflow.start(checkUserThenCreateNftWF, {
-      args: [userDTO],
-      taskQueue: "mint",
-      workflowId: workFlowId,
-    });
-    this.myResponse(res, 200, workFlowId, "set workflow");
-  }
+  checkUserThenCreateNft = async (
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ) => {
+    try {
+      const idToken = req.headers["id-token"]!;
+      const userDTO = new UserDTO();
+      userDTO.publicKey = await getWalletPublicKeyFromIdToken(
+        idToken.toString()
+      );
+      userDTO.email = await getEmailFromIdToken(idToken.toString());
```

Comment:
are these property extraction from jwt token really async functions that need await?

Reply:
since these validations are in middleware, we could just set these values on request object in middleware and just use them here